packages/api/src/services/tags.service.ts

```diff
@@ -13,4 +15,13 @@ export class TagsService {
     async findAll(userId: number): Promise<TransactionTag[]> {
         return this.tagRepository.findBy({ userId });
     }
+
+    async create(req: CreateTagRequest): Promise<TransactionTag> {
+        if (await this.tagRepository.findOneBy(req)) {
+            throw new BadRequestException(ERROR_TAG_NAME_ALREADY_ADDED_TO_ACCOUNT);
+        }
+        const createdTag = this.tagRepository.create(req);
+        await this.tagRepository.insert(createdTag);
+        return await this.tagRepository.findOneBy(createdTag);
```

Comment:
return === no await, its convention to not await the return when we want to return a promise. This can sometimes change inside a try catch but thats for a different time

```suggestion
        return this.tagRepository.findOneBy(tag);
```